### PR TITLE
[test] Reduce loop wait time

### PR DESCRIPTION
**Pull request by aledbf**
## Description

- Refactor loops and delays in the integration suite.
- Ensure the snapshot file from remote storage can be written
 
## How to test
- Integration tests should pass
- Open a workspace using a prebuild or backup (no PVC)

## Release Notes
```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=workspace
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by aledbf** [comment]
/werft run with-clean-slate-deployment=true with-large-vm=true

**Review by utam0k — APPROVED** [review]
Thanks :100:

File: components/ws-daemon/pkg/content/initializer.go [hunk 1]
```diff
@@ -390,26 +390,35 @@ func (rs *remoteContentStorage) Download(ctx context.Context, destination string
...
-	out, err = cmd.CombinedOutput()
+	out, err := cmd.CombinedOutput()
 	if err != nil {
 		log.WithError(err).WithField("out", string(out)).Error("unexpected error downloading file")
 		return true, xerrors.Errorf("unexpected error downloading file")
 	}
 
+	tempFile, err = os.Open(tempFile.Name())
+	if err != nil {
+		return true, xerrors.Errorf("unexpected error downloading file")
```
Comment (kylos101) [inline components/ws-daemon/pkg/content/initializer.go:416]: Nit: we should try to change error messages to be more distinct. For example, so it's different from L411 and L416.

```suggestion
		return true, xerrors.Errorf("unexpected error opening temp file")
```

**Review by kylos101 — COMMENTED** [review]
Hey @aledbf , just a couple questions and a nit, not blocking (especially considering this is merged).

**Pull request merged**